concrete/src/Page/Sitemap/Element/SitemapPage.php

```diff
@@ -0,0 +1,307 @@
+<?php
+
+namespace Concrete\Core\Page\Sitemap\Element;
+
+use Concrete\Core\Error\UserMessageException;
+use Concrete\Core\Page\Page;
+use Concrete\Core\Url\UrlInterface;
+use DateTime;
+use SimpleXMLElement;
+
+class SitemapPage extends SitemapElement
+{
+    /**
+     * The page.
+     *
+     * @var \Concrete\Core\Page\Page
+     */
+    protected $page;
+
+    /**
+     * The URL of the page.
+     *
+     * @var \Concrete\Core\Url\UrlInterface
+     */
+    protected $url;
+
+    /**
+     * The last modification date/time.
+     *
+     * @var \DateTime|null
+     */
+    protected $lastMod;
+
+    /**
+     * The page change frequency.
+     *
+     * @var string
+     */
+    protected $changeFrequency;
+
+    /**
+     * The page priority.
+     *
+     * @var string
+     */
+    protected $priority;
+
+    /**
+     * The pages in alternative languages mapped to this page.
+     *
+     * @var \Concrete\Core\Page\Sitemap\Element\SitemapPageAlternativeLanguage[]
+     */
+    protected $alternativeLanguages = [];
+
+    /**
+     * Should this page be skipped?
+     *
+     * @var bool
+     */
+    protected $skip = false;
+
+    /**
+     * @param Page $page the page
+     * @param UrlInterface $url the URL of the page
+     * @param \DateTime|null the last modification date/time
+     * @param string $changeFrequency the page change frequency
+     * @param string $priority the page priority
+     * @param null|DateTime $lastMod
+     */
+    public function __construct(Page $page, UrlInterface $url, DateTime $lastMod = null, $changeFrequency = '', $priority = '')
+    {
+        $this->page = $page;
+        $this->setUrl($url);
+        $this->setLastMod($lastMod);
+        $this->setChangeFrequency($changeFrequency);
+        $this->setPriority($priority);
+    }
+
+    /**
+     * Get the page.
+     *
+     * @return \Concrete\Core\Page\Page
+     */
+    public function getPage()
+    {
+        return $this->page;
+    }
+
+    /**
+     * Get the URL of the page.
+     *
+     * @return \Concrete\Core\Url\UrlInterface
+     */
+    public function getUrl()
+    {
+        return $this->url;
+    }
+
+    /**
+     * Set the URL of the page.
+     *
+     * @param \Concrete\Core\Url\UrlInterface $url
+     *
+     * @return $this
+     */
+    public function setUrl(UrlInterface $url)
+    {
+        $this->url = $url;
+
+        return $this;
+    }
+
+    /**
+     * Get the last modification date/time.
+     *
+     * @return \DateTime|null
+     */
+    public function getLastMod()
+    {
+        return $this->lastMod;
+    }
+
+    /**
+     * Set the last modification date/time.
+     *
+     * @param \DateTime|null $lastMod
```

Comment:
Out of curiosity, is there a reason why you use `\DateTime` in the docblock, but the import `DateTime` in the method type hint?

Reply:
I adopted this approach because:
1. when we generate `/concrete/src/Support/__IDE_SYMBOLS__.php` with the `c5:ide-symbols` command we copy the PHPDoc from the source files, but we do not copy the `use` statements, so the classes defined in that stub file are wrong
2. what would you do when a class is referenced in a PHPDoc but not in actual PHP code? Would you add a `use` statement just for PHPDoc? Or would you use the fully qualified class name only in these cases? And what would you do if afterwards you add PHP code with that class? Would you parse all the PHPDoc to remove the fully qualified names?

Reply:
I hadn't thought of 1). Makes sense!